pkg/logs/scheduler/scheduler.go

```diff
@@ -33,19 +41,74 @@ var (
 type Scheduler struct {
 	sources  *logsConfig.LogSources
 	services *service.Services
+
+	// context controls the lifetime of this service
+	ctx context.Context
+
+	// cancel cancels the context
+	cancel context.CancelFunc
+
+	// events from the workloadmeta store
+	events chan workloadmeta.EventBundle
+
+	// services we have added, so that we can remove them again, indexed by ID
+	addedServices map[workloadmeta.EntityID]*service.Service
 }
 
 // CreateScheduler creates the scheduler.
 func CreateScheduler(sources *logsConfig.LogSources, services *service.Services) {
+	wlms := workloadmeta.GetGlobalStore()
```

Comment:
Your life writing tests will be a bit easier if you store a reference to this in `Scheduler` instead of hardcoding the global store in several places, so you can use `workloadmetatesting.Store` [from here](https://github.com/DataDog/datadog-agent/blob/main/pkg/workloadmeta/testing/store.go).

Reply:
Looks handy, but https://pkg.go.dev/github.com/DataDog/datadog-agent/pkg/workloadmeta/testing#Store.Subscribe says Subscribe is not implemented!

Reply:
(I ended up implementing Subscribe, although not with filtering)